ui-app/client/src/components/Iframe/Iframe.tsx

```diff
@@ -39,26 +39,41 @@ function Iframe(props: ComponentProps) {
 		{},
 		stylePropertiesWithPseudoStates,
 	);
+
+	const shouldRenderIframe = (srcdoc: any, src: any) => {
+		if (srcdoc) return true;
+		if (!src || !src.trim()) return false;
+		try {
+			new URL(src);
+			return true;
+		} catch (e) {
+			return false;
+		}
+	};
+
```

Comment:
Don't write a function, unless you re-use.

```
let shouldRenderIframe = true;
if (!srcdoc?.trim()) {
  try { new URL(src); } catch(err) { shouldRenderIframe = false; }
}
```

Don't complicate, KISS **(Keep It Stupidly Simple)**